### ISSUE TITLE
Allow to handle network issue at consumer level

### ISSUE THREAD

**Issue by HashirHussain** [okta/okta-auth-js#200]
Scenario:

1. User is using the application peacefully.
2. His system goes to sleep due to inactivity hence network got disconnect.
3. User resumes his system after few hours.
4. He started using the application again.
5. He found his application got hanged(not able to click anywhere) and he can't resume his work until he refresh the page.
6. He found so many  `/authorize` api is failing due to network issue.

Expectation: There must be some exception handler provided by library so the developer can handle the catch and show a friendly message like `Something went wrong, please re-login to continue`.

**Comment by swiftone** [comment]
Thanks for the report - For clarity, were you using the okta-auth-js library directly, or via one of our wrapping SDKs?

Do you have a simple example to recreate the problem, or the series of calls that got you into that scenario?

**Comment by HashirHussain** [comment]
Yes I'm using `@okta/okta-react` version: `1.0.2`. The above problem occurs if user is disconnected from the network for the long time.

**Comment by swiftone** [comment]
Thanks for the information.  We're considering how best to handle this case - would you expect to get numerous network failures to your exception handler?

Is your request only for the exception handler, or to have a graceful recovery instead of requiring a page reload?

**Comment by swiftone** [comment]
Similar error handling discussed at https://github.com/okta/okta-oidc-js/pull/236 (vue) and https://github.com/okta/okta-oidc-js/issues/418 (react)
